modules/options/src/main/scala/scala/build/actionable/ActionableHandler.scala

```diff
@@ -0,0 +1,47 @@
+package scala.build.actionable
+
+import scala.build.Ops._
+import scala.build.errors.{BuildException, CompositeBuildException}
+import scala.build.options.BuildOptions
+
+trait ActionableHandler {
+
+  /** Type of option used to generate actionable diagnostic
+    */
+  type V
+
+  /** Returned type of actionable diagnostics
+    */
+  type A
```

Comment:
Don't we need to add a type bound for the `A` instances returned by `actionableDiagnostic` and `createActionableDiagnostics` to be useful?
```suggestion
  type A <: ActionableDiagnostic
```